### ISSUE TITLE
Databricks dialect doesn't support TRACK HISTORY on table creation

### ISSUE THREAD

**Issue by TheCleric** [sqlfluff/sqlfluff#5947]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When creating a table with the `TRACK HISTORY` option in Databricks, sqlfluff marks it as unparsable.

### Expected Behaviour

The sql parses correctly per https://docs.databricks.com/en/delta-live-tables/sql-ref.html#change-data-capture-with-sql-in-delta-live-tables

### Observed Behaviour

An error message like the following occurs:

`Line 42, Position 1: Found unparsable section: 'TRACK HISTORY ON\n    Delivery,\n    A...'`

### How to reproduce

Here is a minimal SQL example:

```sql
CREATE OR REFRESH STREAMING TABLE address;

APPLY CHANGES INTO LIVE.address
FROM CLOUD_FILES(
    's3://bucket/Address',
    'parquet'
  )
KEYS (AddressKey)
APPLY AS DELETE WHEN
  Op = 'D'
SEQUENCE BY
  DmsTimestamp
COLUMNS * EXCEPT
  (Op, DmsTimestamp, _rescued_data)
STORED AS
  SCD TYPE 2
TRACK HISTORY ON
    Delivery,
    Alternate1,
    Alternate2
;
```

### Dialect

Databricks

### Version

sqlfluff, version 3.0.7
Python 3.10.12

### Configuration

```cfg
[sqlfluff]
dialect = databricks
templater = placeholder
max_line_length = 120
exclude_rules = CP02
large_file_skip_byte_limit = 0

[sqlfluff:indentation]
tab_space_size = 2
indented_joins = True
indented_ctes = True
indented_using_on = True
indented_on_contents = True
# indented_then = True
# indented_then_contents = True
allow_implicit_indents = True
template_blocks_indent = True

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:templater:placeholder]
param_regex = \${(?P<param_name>[-\w.]+)}
ese-pipeline.catalog = ese_dev
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
